tests/plugins/test_gen_protobuf.py

```diff
@@ -0,0 +1,255 @@
+import os
+from unittest import TestCase
+from tempfile import TemporaryDirectory, NamedTemporaryFile
+from nose2.tools import params
+
+from aac.plugins.gen_protobuf.gen_protobuf_impl import gen_protobuf, _convert_camel_case_to_snake_case, _generate_protobuf_template_details_from_data_and_enum_models
+
+
+class TestGenerateProtobufPlugin(TestCase):
+
+    def test_gen_protobuf(self):
+        with TemporaryDirectory() as temp_dir, NamedTemporaryFile("w") as temp_arch_file:
+            temp_arch_file.write(TEST_ARCH_YAML_STRING)
+            temp_arch_file.seek(0)
+
+            gen_protobuf(temp_arch_file.name, temp_dir)
+
+            # Assert each data message has its own file.
+            generated_file_names = os.listdir(temp_dir)
+            self.assertEqual(5, len(generated_file_names))
+
+            # Assert each expected file is present
+            self.assertIn("data_a.proto", generated_file_names)
+            self.assertIn("data_b.proto", generated_file_names)
+            self.assertIn("data_c.proto", generated_file_names)
+            self.assertIn("message_metadata_data.proto", generated_file_names)
+            self.assertIn("message_type.proto", generated_file_names)
+
+            # Assert data_a.proto contents
+            with open(os.path.join(temp_dir, "data_a.proto")) as data_a_proto_file:
```

Comment:
I think we could extract a lot of the duplication here into a separate function. Maybe something like:

```python
def check_generated_file_contains(path: str, contents: list[str]) -> None:
    with open(path) as generated_file:
        generated_file_contents = generated_file.read()
        for string in contents:
            self.assertIn(string, generated_file_contents)
```

And then we could call it like this:

```python
check_generated_file_contains(os.path.join(temp_dir, "data_a.proto"), [
    "import \"message_metadata_data.proto\"",
    "import \"message_type.proto\"",
    "MessageMetadataData metadata",
    # etc
])
```

Especially since this is a common pattern we have for testing, we may want to make this available for other tests, as well.

Reply:
I agree. We should make some testing utilities.

I'll make a story for us to implement some testing utilities; be sure to think about some other functions that'd be helpful.

Reply:
The related issue is #158. The topic is a draft for now.